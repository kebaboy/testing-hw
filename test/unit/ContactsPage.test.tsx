import '@testing-library/jest-dom';
import React from "react";
import { basename, renderApp } from "../utils";
import { Application } from "../../src/client/Application";
import { screen } from '@testing-library/react';

describe('Страница контактов', () => {
    it('должна иметь статическое содержание', () => {
        const {container} = renderApp(<Application/>, [`${basename}/contacts`]);

        expect(container).toMatchSnapshot('Страница контактов');
    })

    it('ссылка "Contacts" в навигации должна быть активной', () => {
        const { container } = renderApp(<Application />, [`${basename}/contacts`]);
        const link = screen.getAllByRole('link', { name: /contacts/i })[0];

        // screen.logTestingPlaygroundURL();
        expect(link).toBeInTheDocument();
        expect(link).toHaveClass('active');
    })
})