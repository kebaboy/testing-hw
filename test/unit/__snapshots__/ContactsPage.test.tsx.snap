// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Страница контактов должна иметь статическое содержание: Страница контактов 1`] = `
<div>
  <div
    class="Application"
  >
    <nav
      class="navbar navbar-expand-sm navbar-light bg-light"
    >
      <div
        class="container"
      >
        <a
          class="Application-Brand navbar-brand"
          href="/hw/store"
        >
          Kogtetochka store
        </a>
        <button
          aria-label="Toggle navigation"
          class="Application-Toggler navbar-toggler"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="Application-Menu collapse navbar-collapse"
        >
          <div
            class="navbar-nav"
          >
            <a
              class="nav-link"
              href="/hw/store/catalog"
            >
              Catalog
            </a>
            <a
              class="nav-link"
              href="/hw/store/delivery"
            >
              Delivery
            </a>
            <a
              aria-current="page"
              class="active nav-link"
              href="/hw/store/contacts"
            >
              Contacts
            </a>
            <a
              class="nav-link"
              href="/hw/store/cart"
            >
              Cart
            </a>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="container pt-4"
    >
      <div
        class="Contacts"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <h1>
              Contacts
            </h1>
            <p>
              Have a question about our scratchers or need help placing an order? Don't hesitate to reach out to us! Our dedicated team is here to provide you with top-notch service and support.
            </p>
            <p>
              Our friendly representatives are available during business hours to assist you with any inquiries you may have.
            </p>
            <p>
              At our store, customer satisfaction is our priority, and we're committed to ensuring you have a smooth and enjoyable shopping experience. Reach out to us today – we're here to help make your cat's scratching dreams a reality!
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
